docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md

```diff
@@ -676,6 +694,40 @@ individually.
 
 </aside>
 
+### Version-aware patches
+
+In some cases the ClusterClass authors want a patch to be computed according to the Kubernetes version in use.
+
+While this is not a problem "per se" and it does not differ from writing any other patch, it is important 
+to keep in mind that there could be different Kubernetes version in a Cluster at any time, all of them
+accessible via built in variables:
+ 
+- `builtin.cluster.topology.version` defines the Kubernetes version from `cluster.topology`, and it acts
+  as the desired Kubernetes for the entire cluster. However, during an upgrade workflow it could happen that
+  some objects in the Cluster are still at older version.
+- `builtin.controlPlane.version`, represent the desired version for the control plane object; usually this
+  version changes immediately after `cluster.topology.version` is updated (unless there are other operations
+  in progress preventing the upgrade to start).
+- `builtin.machineDeployment.version`, represent the desired version for each specific MachineDeployment object;
+  this version changes only after the upgrade for the control plane is completed, and in case of many
+  MachineDeployments in the same cluster, they are upgraded sequentially.
+
+This info should provide the bases for developing version-aware patches, allowing the patch author to determine when a
+patch should adapt to the new Kubernetes version by choosing one of the above variables. In practice the
+following rules applies to the most common use cases:
+
+- When developing a version-aware patch for the control plane, `builtin.controlPlane.version` must be used.
+- When developing a version-aware patch for MachineDeployments, `builtin.machineDeployment.version` must be used.
+
+**Tips & Tricks**:
+
+Sometimes users need to define variables to be used by version-aware patches, and also in this case it is important
+to keep in mind that there could be different Kubernetes version in a Cluster at any time.
+
+A simple approach to solve this problem is to define a map of version-aware variables, with the key of each item
+being the Kubernetes version. Patch could then use the proper builtin variables as a lookup entry to fetch 
+the corresponding values for the Kubernetes version in use by each object.
+
```

Comment:
Sounds good to me but I wonder if it would be easier to understand if we add a concrete example. I already have an example for map lookups based on md class (grep for mdConfig). If we duplicate it here but based on the version it would probably be a lot easier to understand for users.

Reply:
Let's start to merge the description, I can open an issue to add examples when we are back from KubeCon